src/components/Showcase/index.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react'
+import { RestaurantsContext } from '@providers/Restaurants'
+import Loading from '@components/Loading'
+import { Tag, Env, Title, List } from './styles'
+import Item from './Item'
+import { ShowcaseProps } from '@utils/types'
+
+const Showcase: React.FunctionComponent<ShowcaseProps> = ({ title, list, isLoading }) => {
+  const { priceFilter, openFilter } = React.useContext(RestaurantsContext)
+
+  return (
+    <Tag data-testid="showcase">
+      <Env>
+        <Title>
+          { title }
+        </Title>
+        <List>
+          {/* TODO: Improve if nesting logic */ }
+          { list.map((item, key) => {
+            if(priceFilter.length > 0 || openFilter) {
+              if(openFilter) {
+                if(openFilter && !item.is_closed) {
+                  if(priceFilter.length > 0) {
+                    if(priceFilter.includes(item.price)) {
+                      return (<Item item={item} key={key} />)
+                    } else {
+                      return false
+                    }
+                  } else {
+                    return (<Item item={item} key={key} />)
+                  }
+                }
+              } else {
+                if(priceFilter.includes(item.price))
+                  return (<Item item={item} key={key} />)
+                if(priceFilter.length === 0)
+                  return (<Item item={item} key={key} />)
+                return false
+              }
+            } else {
+              return (<Item item={item} key={key} />)
+            }
```

Comment:
I know that there's a TODO on this this item but do you think you can show me, in a comment below, how you would make this cleaner? There are two main issues in here that I hope could be addressed: 

1) There really are only 2 leafs in this logic flow but there are 7 leafs in your implementation of the branching logic. 
2) This is extremely hard to grok what is going on at first glance.

Reply:
Looking at it with a fresh mind now, it could be all replaced with the following logic:

```suggestion
           if(
              (!openFilter || openFilter === !item.is_closed) && 
              (priceFilter.includes(item.price) || priceFilter.length === 0)
            ) {
              return (<Item item={item} key={key} />)
            } else {
              return false
            }
```

Reply:
Awesome! This is much more clear what's going on in here 😄